lib/std/sort.zig

```diff
@@ -502,6 +509,267 @@ test "binarySearch" {
     );
 }
 
+/// Returns the index of the first element in `items` greater than or equal to `key`,
+/// or `items.len` if all elements are less than `key`.
+///
+/// `items` must be sorted in ascending order with respect to `compareFn`.
+///
+/// O(log n) complexity.
+pub fn lowerBound(
+    comptime T: type,
+    key: anytype,
+    items: []const T,
+    context: anytype,
+    comptime lessThan: fn (context: @TypeOf(context), lhs: @TypeOf(key), rhs: T) bool,
+) usize {
+    var left: usize = 0;
+    var right: usize = items.len;
+
+    while (left < right) {
+        const mid = left + (right - left) / 2;
+        if (lessThan(context, items[mid], key)) {
```

Comment:
@schmee This looks wrong. The type of `items[mid]` is `T`, not `@TypeOf(key)`. I think this only passes the tests because none of them end up with different types here.

Reply:
Yup! I actually ran into this myself the last few days, if no one else makes a PR to fix it I'll get around to it eventually 👍

Reply:
Tests should be changed to use non coerce-able types to prevent this.